### PR TITLE
Ivan features

### PR DESCRIPTION
**Pull request by risticivan12**

### CODE REVIEW

File: src/main/java/com/internship/frejaeidjmeterplugin/jmeter/sampler/impl/FrejaEidPluginSampler.java [hunk 1]
```diff
@@ -37,19 +45,44 @@ public void setRequests(String requests) {
         setProperty("requests", requests);
     }
 
+    public String getEmailFilePath() {
```
Comment (risticivan12) [inline src/main/java/com/internship/frejaeidjmeterplugin/jmeter/sampler/impl/FrejaEidPluginSampler.java]: this should be remove , also this three methods bellow, i forgot that part of code.

File: src/main/java/com/internship/frejaeidjmeterplugin/jmeter/email/impl/ConfirmEmailService.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            }
+            return lastEmailMessage;
+        } catch (MessagingException ex) {
+            throw new FrejaEidException("Failed to get last email message from inbox.", ex);
+        }
+    }
+
+    private void closeEmailFolderAndStore() throws FrejaEidException {
+        try {
+         //   TestLogger.info("Closing email folder and store. ");
```
Comment (aleksandarmarkovic-verisec) [inline src/main/java/com/internship/frejaeidjmeterplugin/jmeter/email/impl/ConfirmEmailService.java]: Remove commented lines

File: src/main/java/com/internship/frejaeidjmeterplugin/jmeter/email/impl/ConfirmEmailService.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    public ConfirmEmailService() throws Exception {
+        httpService = new HttpService(EnviromentSettings.getWebSocketKeystorePath(), EnviromentSettings.getKeystorePassword());
+    }
+
+    @Override
+    public void openAndConfirmLinkFromLastEmailMessage() throws FrejaEidException, IOException, MessagingException, InterruptedException {
+        try {
+            connectToEmailServer(EnviromentSettings.getUserEmail(), EnviromentSettings.getUserEmailAccountPassword());
+            String confirmLink = getConfirmRestoreLinkFromEmailContent(getEmailMessageContent(getConfirmRestoreEmailMessage()));
+          //  TestLogger.info("Click on link for confirming email: " + confirmLink);
```
Comment (aleksandarmarkovic-verisec) [inline src/main/java/com/internship/frejaeidjmeterplugin/jmeter/email/impl/ConfirmEmailService.java]: Remove commented lines in whole class

File: src/main/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/MobileClientService.java [hunk 4]
```diff
@@ -1,34 +1,81 @@
...
+
+    public ResponseStatus declineTransaction(MobileClientApi mobileClient, String reference) {
+        return mobileClient.declineTransaction(reference);
+    }
+
+    public void closeConnection(MobileClientApi mobileClient) {
         mobileClient.closeConnection();
     }
 
+    public void populateMobileClients(List<String> emails) {
```
Comment (aleksandarmarkovic-verisec) [inline src/main/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/MobileClientService.java]: maybe its better to call this method - connectMobileClientsWithEmails

File: src/main/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/MobileClientService.java [hunk 5]
```diff
@@ -1,34 +1,81 @@
...
 
     public MobileClientService() throws FrejaEidException, Exception {
-        mobileClient = new MobileClient(EnviromentSettings.getWebSocketAdress(), EnviromentSettings.getWebSocketKeystorePath(), EnviromentSettings.getKeystorePassword());
+        confirmEmailService = new ConfirmEmailService();
+        emails = new HashMap<>();
+        mobileClients = new HashMap<>();
     }
 
     static {
         JsonServiceFactory.registerJsonServiceProvider(new JsonService());
```
Comment (aleksandarmarkovic-verisec) [inline src/main/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/MobileClientService.java]: We already have this

File: src/main/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/MobileClientService.java [hunk 6]
```diff
@@ -1,34 +1,81 @@
...
+    }
+
+    public void closeConnection(MobileClientApi mobileClient) {
         mobileClient.closeConnection();
     }
 
+    public void populateMobileClients(List<String> emails) {
+    }
+
+    public MobileClient getMobileClient(String email) {
```
Comment (aleksandarmarkovic-verisec) [inline src/main/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/MobileClientService.java]: getMobileClientForEmail

File: src/main/java/com/internship/frejaeidjmeterplugin/jmeter/settings/EmailSettings.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class EmailSettings {
+
+    private static HashMap<Integer, String> emails;
+    private static List<String> passwords;
+
+    static {
+        emails = new HashMap<>();
```
Comment (aleksandarmarkovic-verisec) [inline src/main/java/com/internship/frejaeidjmeterplugin/jmeter/settings/EmailSettings.java:21]: Why do we need emails map also here? I think we have it in MobileClientService class.

File: src/main/java/com/internship/frejaeidjmeterplugin/jmeter/settings/EnviromentSettings.java [hunk 8]
```diff
@@ -16,6 +16,9 @@
     private static String relyingPartyKeystorePath;
     private static String serviceAddress;
     private static String webSocketAdress;
+    private static String USER_EMAIL = "petarpetrovic123.test@gmail.com";
```
Comment (aleksandarmarkovic-verisec) [inline src/main/java/com/internship/frejaeidjmeterplugin/jmeter/settings/EnviromentSettings.java]: I guess this should be deleted when we implement reading from txt file

**Review by S-Ptr — APPROVED** [review]

**Pull request closed without merging**